File: Java/Numbers/program49.java
//Accept number from user and check whether number is prime or not

import java.util.*;

class Number
{
    public int iNo;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);
        
        System.out.println("Enter the number");
        iNo = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("The numbers is: "+this.iNo);
    }

    public boolean CheckPrime()
    {
        int iCnt = 0;
        boolean bFlag = true;

        for(iCnt = 2; iCnt <=(iNo/2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                bFlag = false;
                break;
            }
        }
        return bFlag;
    }
}

class program49
{
    public static void main(String arg[])
    {
        boolean bRet;

        Number obj = new Number();

        obj.Accept();
        obj.Display();

        bRet = obj.CheckPrime();
        if(bRet == true)
        {
            System.out.println("The number is prime number");
        }
        else
        {
            System.out.println("The number is not a prime number");
        }
    }
}

File: C++/Number/program49.cpp
// Accept number from user and check whether the number is perfect or not

#include <iostream>
using namespace std;

class Number
{
public:
    int iNo;

    void Accept() // Setter
    {
        cout << "Enter the number" << endl;
        cin >> iNo;
    }

    void Display() // Getter
    {
        cout << "The number is: " << this->iNo << endl;
    }

    bool CheckPerfect()
    {
        int iCnt = 0;
        int iSum = 0;

        if (iNo < 0)
        {
            iNo = -iNo;
        }

        for (iCnt = 1; ((iCnt <= (iNo /2)) && (iSum <= iNo)); iCnt++)
        {
            if ((iNo % iCnt) == 0)
            {
                iSum = iSum + iCnt;
            }
        }

        if (iSum == iNo)
        {
            return true;
        }
        else
        {
            return false;
        }
    }
};

int main()
{
    Number nobj;

    bool bRet;

    nobj.Accept();
    nobj.Display();

    bRet = nobj.CheckPerfect();
    if (bRet == true)
    {
        cout << "The number is perfect number" << endl;
    }
    else
    {
        cout << "The number is not a perfect number" << endl;
    }

    return 0;
}
